explainaboard/analysis/analyses.py

```diff
@@ -269,12 +269,8 @@ def perform(
             )
 
             n_samples = len(bucket_collection.samples)
-            bucket_performance = BucketPerformance(
-                n_samples=n_samples,
-                bucket_samples=subsampled_ids,
-                bucket_interval=bucket_collection.interval,
-                bucket_name=bucket_collection.name,
-            )
+
+            performances: dict[str, Performance] = {}
```

Comment:
Do we need to require that the `performances` need to maintain insertion order?

Reply:
It would be nice to maintain this property. It seems that as of Python 3.7, dicts are guaranteed to maintain insertion order: https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6

Reply:
I don't think assuming any order is semantically good idea if we are using hashmap (dict) even if the language supports it. If the order of names is meaningful for someone, arranging the order should be the responsibility of callers.